src/cloud-api-adaptor/test/provisioner/gcp/cluster.go

```diff
@@ -0,0 +1,346 @@
+	if err != nil {
+		return nil, fmt.Errorf("invalid node_count: %v", err)
+	}
+
+	return &GKECluster{
+		clusterName:        defaults["cluster_name"],
+		clusterVersion:     defaults["cluster_version"],
+		clusterMachineType: defaults["cluster_machine_type"],
+		credentials:        defaults["credentials"],
+		nodeCount:          nodeCount,
+		ProjectID:          defaults["project_id"],
+		Zone:               defaults["zone"],
+		cluster:            nil,
+	}, nil
+}
+
+// Apply basic labels to worker nodes
+func (g *GKECluster) ApplyNodeLabels(ctx context.Context) error {
+	kubeconfigPath, err := g.GetKubeconfigFile(ctx)
+	if err != nil {
+		return err
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to build kubeconfig: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create clientset: %v", err)
+	}
+
+	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	for _, node := range nodes.Items {
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			n, err := clientset.CoreV1().Nodes().Get(ctx, node.Name, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get node: %v", err)
+			}
+
+			n.Labels["node.kubernetes.io/worker"] = ""
+			n.Labels["node-role.kubernetes.io/worker"] = ""
+			_, err = clientset.CoreV1().Nodes().Update(ctx, n, metav1.UpdateOptions{})
+			return err
+		})
+		if err != nil {
+			return fmt.Errorf("Failed to label node %s: %v\n", node.Name, err)
+		}
+		log.Infof("Successfully labeled node %s\n", node.Name)
+	}
+	return nil
+}
+
+// CreateCluster creates the GKE cluster
+func (g *GKECluster) CreateCluster(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Hour)
+	defer cancel()
+
+	srv, err := container.NewService(
+		ctx, option.WithCredentialsFile(g.credentials),
+	)
+	if err != nil {
+		return fmt.Errorf("GKE: container.NewService: %v", err)
+	}
+
+	cluster := &container.Cluster{
+		Name:                  g.clusterName,
+		InitialNodeCount:      g.nodeCount,
+		InitialClusterVersion: g.clusterVersion,
+		Network:               "test-provisioning-e2e",
+		NodeConfig: &container.NodeConfig{
+			MachineType: g.clusterMachineType,
+			ImageType:   "UBUNTU_CONTAINERD", // Default CO OS has a ro fs.
+		},
+	}
+
+	req := &container.CreateClusterRequest{
+		Cluster: cluster,
+	}
+
+	op, err := srv.Projects.Zones.Clusters.Create(
+		g.ProjectID, g.Zone, req,
+	).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("GKE: Projects.Zones.Clusters.Create: %v", err)
+	}
+
+	log.Infof("GKE: Cluster creation operation: %v\n", op.Name)
+
+	g.cluster, err = g.WaitForClusterActive(ctx, 30*time.Minute)
+	if err != nil {
+		return fmt.Errorf("GKE: Error waiting for cluster to become active: %v", err)
+	}
+
+	wd, err := os.Getwd()
```

Comment:
Printing wd only for debugging sake?

Reply:
yes, leftover, nice catch. I will remove with force push.